docs/reference/contributing/guidelines/workflow.md

```diff
@@ -50,53 +68,69 @@ Pull requests on GitHub have to meet the following requirements to keep the code
 
 #### GitHub pull requests workflow
 
 Each pull request goes through the following workflow:
 
 <span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/Workflow.png)<span>The workflow of merging a pull request</span></span>
 
-##### Pull request states
+#### Pull request states
 
-Labels that the Mbed OS maintainers add to a pull request represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
+The Mbed OS maintainers add labels to a pull request that represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
+
+Each state is time boxed. In most cases, this is sufficient time to move to an another state.
 
 ##### Reviews
 
-All pull requests must be reviewed. The Mbed OS maintainers determine the most suitable person to review the pull request and tag that person accordingly. If a pull request requires a review from partners, the maintainers tag the corresponding GitHub group instead.
+All pull requests must be reviewed. The Arm Mbed CI bot determines the most suitable person to review the pull request and tags that person accordingly.
+
+Time: 3 days for reviewers to leave feedback after the maintainers add the "needs: review" label.
 
 ##### The CI (Continuous Integration) testing
 
-There are a number of CI systems available. Which CI tests we run against a particular pull request depends on the effect of that pull request to the code base. Irrespective of which CIs tests run, Mbed OS has an all green policy, meaning that all the CI jobs that are triggered must pass before we merge the pull request.
+There are many CI systems available. Which CI tests we run against a particular pull request depends on the effect that pull request has on the code base. Irrespective of which CIs tests run, Mbed OS has an all green policy, meaning that all the CI jobs that are triggered must pass before we merge the pull request.
+
+Time: 1 day for CI to complete and report back results.
+
+##### Work needed
+
+A pull request in the work needed state requires additional work due to failed tests. If a pull request is in this state, our maintainers request changes from the pull request author.
+
+Time: 3 days for the pull request author to respond to the review comments.
 
 ##### Releases
 
-Once we merge a pull request, we tag it with a release. This is the release in which we first publish this pull request. For patch releases, we allow only bug fixes, new targets and enhancements to existing functionality. New features only go to feature releases.
+Once we merge a pull request, we tag it with a release. (This is not applicable for feature pull requests.) This is the release in which we first publish this pull request. For patch releases, we allow only bug fixes, new targets and enhancements to existing functionality. New features only go to feature releases.
```

Comment:
Don't we tag features with the next applicable feature release? Why would we omit that step? Why is this in parenthesis? If we want to omit tagging things that go to feature releases it would be less confusing to word it as:
"When we merge a pull request that will be published in a patch release, we tag that pull request with that patch release version."

Reply:
I haven't noticed us tagging a feature PR with a target release.

+1 for the updated text.

Reply:
> "When we merge a pull request that will be published in a patch release, we tag that pull request with that patch release version."

Will fix